src/hooks/usePreferences.js

```diff
@@ -7,41 +10,50 @@ import { useStripes } from '../StripesContext';
 export default () => {
   const { user, logger } = useStripes();
   const ky = useOkapiKy();
-  const [id, setId] = useState(null);
+
+  // use a ref to store the id of the setting so that it will be available for the next request when we chain.
+  const idRef = useRef(null);
+  const setId = useCallback(newId => { idRef.current = newId; }, []);
+
   const userId = user.user.id;
 
-  const getPreference = useCallback(async ({ scope, key }) => { // eslint-disable-line
+  const getPreference = useCallback(async ({ scope, key, userId: providedUserId }) => { // eslint-disable-line
     // get preferences by querying preferences with the userId, scope, and key...
     // this will return an array that's likely a single item long. We store the item's id in state so that we can update it later.
```

Comment:
```suggestion
    // this will return an array that's likely a single item long. We store the item's id in a ref so that we can update the preference later.
```
Minor - just a comment that would need to be updated given the changes to the implementation.